### PR TITLE
Refactor: Extract common method to check if any ancestor belongs to SyntaxKind

### PR DESCRIPTION
**Pull request by mary-georgiou-sonarsource**

### CODE REVIEW

**Review by gregory-paidis-sonarsource — APPROVED** [review]
LGTM

File: analyzers/src/SonarAnalyzer.CSharp/Common/Walkers/MutedSyntaxWalker.cs [hunk 1]
```diff
@@ -113,18 +113,15 @@ public override void VisitVariableDeclarator(VariableDeclaratorSyntax node)
...
-                isInTryOrCatch = isInTryOrCatch || !HasAncestor(node, SyntaxKind.FinallyClause);
+                isInTryOrCatch = isInTryOrCatch || !node.HasAncestorOfKind(SyntaxKind.FinallyClause);
             }
             else
             {
                 isOutsideTryCatch = true;
             }
         }
-
-        private static bool HasAncestor(SyntaxNode node, SyntaxKind kind) =>
```
Comment (mary-georgiou-sonarsource) [inline analyzers/src/SonarAnalyzer.CSharp/Common/Walkers/MutedSyntaxWalker.cs:127]: it has 0 references in the project.

File: analyzers/src/SonarAnalyzer.CSharp/Helpers/CSharpSyntaxHelper.cs [hunk 2]
```diff
@@ -70,37 +69,22 @@ public static SyntaxNode GetSelfOrTopParenthesizedExpression(this SyntaxNode nod
...
         node.GetSelfOrTopParenthesizedExpression().Parent;
 
+    public static bool HasAncestorOfKind(this SyntaxNode syntaxNode, params SyntaxKind[] syntaxKinds) =>
+        syntaxNode.Ancestors().Any(ancestor => ancestor.IsAnyKind(syntaxKinds));
+
     public static bool IsOnThis(this ExpressionSyntax expression) =>
         IsOn(expression, SyntaxKind.ThisExpression);
 
-    public static bool IsOnBase(this ExpressionSyntax expression) =>
-        IsOn(expression, SyntaxKind.BaseExpression);
```
Comment (mary-georgiou-sonarsource) [inline analyzers/src/SonarAnalyzer.CSharp/Helpers/CSharpSyntaxHelper.cs:77]: this has 0 references in the project.

**Review by pavel-mikula-sonarsource — CHANGES_REQUESTED** [review]
Some cleanup

File: analyzers/src/SonarAnalyzer.VisualBasic/SymbolicExecution/Roslyn/PublicMethodArgumentsShouldBeCheckedForNull.cs [hunk 3]
```diff
@@ -85,7 +85,7 @@ public override void Visit(SyntaxNode node)
         public override void VisitIdentifierName(IdentifierNameSyntax node) =>
             DereferencesMethodArguments |=
                 argumentNames.Contains(node.GetName())
-                && node.HasAncestorOfKind(AwaitExpression,
+                && node.HasAncestor(AwaitExpression,
```
Comment (pavel-mikula-sonarsource) [inline analyzers/src/SonarAnalyzer.VisualBasic/SymbolicExecution/Roslyn/PublicMethodArgumentsShouldBeCheckedForNull.cs:88]: [T0028](https://github.com/SonarSource/sonar-dotnet/issues/9161)
```suggestion
                && node.HasAncestor(
                    AwaitExpression,
```

**Review by pavel-mikula-sonarsource — APPROVED** [review]
LGTM

File: analyzers/src/SonarAnalyzer.VisualBasic/SymbolicExecution/Roslyn/PublicMethodArgumentsShouldBeCheckedForNull.cs (hunk 3 above)
Comment (pavel-mikula-sonarsource) [reply analyzers/src/SonarAnalyzer.VisualBasic/SymbolicExecution/Roslyn/PublicMethodArgumentsShouldBeCheckedForNull.cs:88]: You can fix this in your next PR, I'll merge this one now as I need to nuke the whole helper file

**Pull request merged**